x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/amazonbedrock/client/AmazonBedrockChatCompletionExecutor.java

```diff
@@ -34,16 +35,13 @@ protected AmazonBedrockChatCompletionExecutor(
 
     @Override
     protected void executeClientRequest(AmazonBedrockBaseClient awsBedrockClient) {
-        if (chatCompletionRequest.isStreaming()) {
-            var publisher = chatCompletionRequest.executeStreamChatCompletionRequest(awsBedrockClient);
-            inferenceResultsListener.onResponse(new StreamingChatCompletionResults(publisher));
-        } else {
-            var chatCompletionResponseListener = new AmazonBedrockChatCompletionResponseListener(
-                chatCompletionRequest,
-                responseHandler,
-                inferenceResultsListener
-            );
-            chatCompletionRequest.executeChatCompletionRequest(awsBedrockClient, chatCompletionResponseListener);
+        // Chat completions only supports streaming
+        if (chatCompletionRequest.isStreaming() == false) {
+            inferenceResultsListener.onFailure(CHAT_COMPLETION_STREAMING_ONLY_EXCEPTION);
+            return;
```

Comment:
I'm not too sure of the code paths involved here, but is it possible to reach this line without first going through `TransportInferenceActionProxy.doExecute()`, where we do the same check?